src/Hng.Web/Controllers/AuthController.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Hng.Application.Dto;
+using Hng.Infrastructure.Repository.Interface;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Hng.Web.Controllers
+{
+    [ApiController]
+    [Route("api/v1/auth")]
+    public class AuthController : ControllerBase
+    {
+        private readonly IEmailService _emailService;
+        private readonly ITokenService _tokenService;
+        private readonly IJwtService _jwtService;
+        private readonly IConfiguration _configuration;
+        private readonly IUserRepository _userRepository;
+ 
+
+        public AuthController(IEmailService emailService, ITokenService tokenService, IConfiguration configuration, IUserRepository userRepository, IJwtService jwtService)
+        {
+            _userRepository = userRepository;
+            _emailService = emailService;
+            _tokenService = tokenService;
+            _configuration = configuration;
+            _jwtService = jwtService;
+        }
+
+        [HttpGet("signin-token")]
+        public async Task<IActionResult> RequestSignInToken([FromQuery] string email)
```

Comment:
This is meant to be a POST request because you are creating and saving something to the database